### DISCUSSION TITLE
How to call Postgres database functions from custom schema?

### DISCUSSION THREAD

**Discussion by premjit-develops** [supabase-community/supabase-kt#712] [Q&A]
I'm trying to create a custom schema and have added some functions to it, but I'm unsure how to call these functions properly. This is because rpc does not have schema parameter like get() or from().

**Comment by jan-tennert** [comment]
Seems like this functionality is missing currently. I'll add a parameter in a future version, for now you can follow the library's approach and [set the header yourself (which specifies the schema used)](https://github.com/supabase-community/supabase-kt/blob/master/Postgrest%2Fsrc%2FcommonMain%2Fkotlin%2Fio%2Fgithub%2Fjan%2Fsupabase%2Fpostgrest%2Fexecutor%2FPostgrestHttpExtension.kt#L34-L35)
So it would look something like this:
```kotlin
postgrest.rpc("my_function") {
   headers["Content-Profile"] = "my_schema"
} 
```

**Reply by premjit-develops** [reply]
```
 postgrest.rpc(
            function = "get_all_foods",
            parameters = FoodParameters(userId = userId, foodIds = ids),

        ){
          headers["Content-Profile"] = "foodz"
      }.decodeList<Food>()
```
is this correct?

**Reply by jan-tennert** [reply]
Try and see. It should work.

**Reply by premjit-develops** [reply]
Not working..

   ```
NotFoundRestException: Could not find the function public.get_all_foods(foodIds, userId) in the schema cache (Searched for the function public.get_all_foods with parameters foodIds, userId or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache.)
                                      URL: https://pbzksx.supabase.co/rest/v1/rpc/get_all_foods
                                      Headers: [Authorization=[Bearer eyJhbGciOiJIUzI1NiIsImtpZCI6IkpOWWNEdmhFWXc2SD
```

**Reply by jan-tennert** [reply]
Yea, seems like the custom headers are currently ignored by the RPC request. I'll publish `2.6.1` with a fix. After this it should work.
A proper parameter will get introduced in `2.7.0`.

**Reply by jan-tennert** [reply]
Can you try out version 2.6.1-dev?

**Reply by premjit-develops** [reply]
Dev version `2.6.1-dev` is working perfectly.
